src/main/java/org/exschool/pocketworld/dao/AbstractJpaDAO.java

```diff
@@ -0,0 +1,45 @@
+package org.exschool.pocketworld.dao;
+
+import java.io.Serializable;
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+public abstract class AbstractJpaDAO<T extends Serializable> {
```

Comment:
Do you mean something like in this article ?  http://www.baeldung.com/2011/12/08/simplifying-the-data-access-layer-with-spring-and-java-generics/  (2.2 The Abstract Hibernate DAO)